### ISSUE TITLE
Greets!  I love this thing.  Could it work for objects and not just bones?

### ISSUE THREAD

**Issue by spiraloid** [TheDuckCow/pose-tools#1]

**Comment by TheDuckCow** [comment]
Thanks for the idea submission - that's actually an interesting idea. I haven't updated this code in quite awhile (much of it has essentially been picked up in the [pose thumbnails addon](https://github.com/jasperges/pose-thumbnails)), but I'll certainly consider this if I get the time.

Just to be clear, you are referring to mostly scale/location/rotation changes of single objects I presume, and copying / applying those transformations to objects at different points in time? 

Side note to self: if this were done, would have to either be limited to just copying "poses" (transformations for a given moment in time), or would have to invest some amount of time to recreate from scratch an object-version of a pose library data container.

**Comment by spiraloid** [comment]
yeah, I want it for bones, transforms, slider values, colors etc.  everything.  ideally it would query the same thing that the active keying set has and use it.   this is how https://www.studiolibrary.com/  works for maya.  when you make the pose, it stores the selection.  Double click re-selects the contents of the pose.  MMB drag fades in the values in the pose to whatever is selected.    very powerful.  it was pretty much my main animation tool.

**Comment by spiraloid** [comment]
btw.  One could theoretically save/load the poses by key/value in a .json
file but I think alembic would be better as you could store moving
presets...... the UI could mirror a presence inferred folder/file structure
that followed a rigid naming convention.....  so long as the control names
matched in the key string you lerp to a keyframe value in the .abc.

the controls could be applied to the selected objects.  imagine selecting a
crowd of characters hands and applying the hand/fist.abc preset.  the
current values on each bone lerp to the control names value on frame 0 of
the .abc file.

far more valuable feature than thumbnails IMHO.

hands/fist.abc
hands/flat.abc
hands/pointing.abc
hands/fu.abc
hands/thumbs up.abc
viseme/MBP.abc
viseme/OH.abc
viseme/EEE.abc
viseme/FF.abc
viseme/Lth.abc
viseme/KDN.abc
viseme/OO.abc
crowd/bodypile.abc
crowd/line.abc
crowd/sidewalk.abc
crowd/panic.abc
crowd/circle.abc
crowd/bus.abc

**Comment by spiraloid** [comment]
btw:

http://bleaf.tumblr.com/post/179187460274/couldnt-resist-heres-my-startup-file-hotkeys

**Comment by TheDuckCow** [comment]
Thanks for sharing the additional details, indeed I've seen that video/tool as a reference before. I agree with the value prop, not sure if the foundation would agree with the workarounds needed to make an all-encompasing data structure for everything, but a sort of manual override could be done to make it happen. Good idea around json data - though even that likely wouldn't be required, as you can store arbitrary attributes to blender datablocks already (it's more about where to place these different datablocks that gets messy, but doable). 

Focused on a few other things right now, but if I have some time I'll come back to this. In the meantime, I do still recommend the.

Reselecting the relevant content via double clicking (or some other method) is also an interesting and feasible thing, this could be added to the thumbnail repo as well potentially as another operator or hotkey.
